src/MerchantCenter/MerchantStatuses.php

```diff
@@ -166,26 +167,25 @@ public function maybe_refresh_status_data( bool $force_refresh = false ): void {
 			throw new Exception( __( 'No Merchant Center account connected.', 'google-listings-and-ads' ) );
 		}
 
-		// Update MC product stats and issues page by page.
 		$this->mc_statuses = [];
-		$page_token        = null;
-		do {
-			$response = $this->container->get( Merchant::class )->get_productstatuses( $page_token );
-			$statuses = $this->filter_valid_statuses( $response->getResources() );
-			$this->refresh_product_issues( $statuses );
-			$this->sum_status_counts( $statuses );
 
-			$page_token = $response->getNextPageToken();
-		} while ( $page_token );
+		// Update account-level issues.
+		$this->refresh_account_issues();
 
-		$this->update_mc_statuses();
-		$this->update_product_mc_statuses();
+		// Update MC product issues and tabulate statistics in batches.
+		$chunk_size = 5000;
```

Comment:
Do we want to have a helper function/filter for this batch size, in case there are sites that struggle. So we can provide them with a code snippet to adjust it.

Reply:
🧠  98b82ec56551a0e8e373254bdadb07649f12defd